### DISCUSSION TITLE
V5 版配置文件过度精简导致无法按用户路由

### DISCUSSION THREAD

**Discussion by Mukou-Aoi** [v2fly/v2ray-core#1944] [General]
V5 版的配置文件对入站协议设置进行了大幅削减，VMess 的入站配置只剩下了一个 users 数组，这导致 V4 版配置文件中用于识别用户的 email 和用户等级 level 均无法配置。路由功能中 RuleObject 的 "users" (V4 版）/ "userEmail" (V5 版）可以用来匹配用户实现按用户路由，然而因为 V5 版配置文件的入站配置无法设置 email 使得在使用 V5 版配置文件时此功能实质不可用。

按用户路由功能用途是很广泛的，并不只是对有多用户的服务器才有用。我的代理服务器只有我一个人在使用，存在多个入站方式（TLS，websocket，kcp...）和多个出站（IPv4，IPv6，转发到其它代理）实现区分出站用的就是按照用户（email）路由的方法，如果按用户路由不可用，替代方案需要配置大量的入站并按照入站 tag 路由，会导致配置文件长度倍增（大量重复内容）难以维护。

V4 版配置文件的入站配置基本都支持 email（VMess，Shadowsocks, Trojan) ，V5 版配置文件中均被删除，而 email 又是按用户路由的唯一手段，精简这个字段是否欠妥？目前使用 V5 版配置文件因为大幅精简而导致部分功能无法配置，而 V4 版配置文件文档又没有新协议（如 VLite）的相关内容而无法配置，处于新旧配置均无法使用全部功能的尴尬状况。

**Comment by mclovin-2k** [comment]
还是先用V4吧。
再等段时间，V5估计就完善了。

**Comment by mkmark** [comment]
see https://github.com/v2fly/v2ray-core/pull/2933 for example of v5 complete config, which supports routing via user

**Reply by Misaka13514** [reply]
> see #2933 for example of v5 complete config, which supports routing via user

I didn't find it there 🤔 It doesn't even have a RoutingObject
